python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py

```diff
@@ -6,87 +6,154 @@
+
+
+@pytest.fixture
+def sling_sqlite_resource(temp_db):
+    return SlingResource(
+        source_connection=SlingSourceConnection(type="file"),
+        target_connection=SlingTargetConnection(
+            type="sqlite", connection_string=f"sqlite://{temp_db}"
+        ),
+    )
+
+
+ASSET_SPEC = AssetSpec(
+    key=["main", "tbl"],
+    group_name="etl",
+    description="ETL Test",
+    deps=["foo"],
+)
 
 
 @pytest.mark.parametrize(
     "mode,runs,expected", [(SlingMode.INCREMENTAL, 1, 3), (SlingMode.SNAPSHOT, 2, 6)]
 )
-def test_build_sling_asset(mode: SlingMode, runs: int, expected: int):
-    with tempfile.TemporaryDirectory() as tmpdir_path:
-        fpath = os.path.abspath(file_relative_path(__file__, "test.csv"))
-        dbpath = os.path.join(tmpdir_path, "sqlite.db")
+def test_build_sling_asset(
+    test_csv: str,
+    temp_db: str,
+    sling_sqlite_resource: SlingResource,
+    mode: SlingMode,
+    runs: int,
+    expected: int,
+):
+    asset_def = build_sling_asset(
+        asset_spec=ASSET_SPEC,
+        source_stream=f"file://{test_csv}",
+        target_object="main.tbl",
+        mode=mode,
+        primary_key="SPECIES_CODE",
+        sling_resource_key="sling_resource",
+    )
 
-        sling_resource = SlingResource(
-            source_connection=SlingSourceConnection(type="file"),
-            target_connection=SlingTargetConnection(
-                type="sqlite", connection_string=f"sqlite://{dbpath}"
-            ),
-        )
-
-        asset_spec = AssetSpec(
-            key=["main", "tbl"],
-            group_name="etl",
-            description="ETL Test",
-            deps=["foo"],
-        )
-        asset_def = build_sling_asset(
-            asset_spec=asset_spec,
-            source_stream=f"file://{fpath}",
-            target_object="main.tbl",
-            mode=mode,
-            primary_key="SPECIES_CODE",
-            sling_resource_key="sling_resource",
-        )
-
-        sling_job = build_assets_job(
-            "sling_job",
-            [asset_def],
-            resource_defs={"sling_resource": sling_resource},
-        )
-
-        counts = None
-        for n in range(runs):
-            res = sling_job.execute_in_process()
-            assert res.success
-            counts = sqlite3.connect(dbpath).execute("SELECT count(1) FROM main.tbl").fetchone()[0]
-        assert counts == expected
-
-
-def test_can_build_two_assets():
-    with tempfile.TemporaryDirectory() as tmpdir_path:
-        fpath = os.path.abspath(file_relative_path(__file__, "test.csv"))
-        dbpath = os.path.join(tmpdir_path, "sqlite.db")
+    sling_job = build_assets_job(
+        "sling_job",
+        [asset_def],
+        resource_defs={"sling_resource": sling_sqlite_resource},
+    )
+
+    counts = None
+    for n in range(runs):
+        res = sling_job.execute_in_process()
+        assert res.success
+        counts = sqlite3.connect(temp_db).execute("SELECT count(1) FROM main.tbl").fetchone()[0]
```

Comment:
refactor the table count as a helper method, and re-used in other tests?

Reply:
I thought about this too but I was worried there's too much indirection

I think it's very simple to see what `select count(1) from main.tbl` does, but refactoring it like this doesn't seem like it really helps much. 

```python
def get_count_from_conn(conn, tbl): 
    return conn.execute(f"SELECT count(1) FROM {tbl}").fetchone()[0]

...much later....
assert get_count_from_conn(conn, 'main.tbl') == 3

vs.

assert conn.execute("SELECT count(1) FROM main.tbl").fetchone()[0] == 3
```

Second one is more more clear to me, even if it's a little ugly.

Reply:
The amount of indirection added here is minimal. But won't block review on this feedback.